### PR TITLE
🛡️ Sentry: Add table-driven tests for sparse_squared_euclidean_distance edge cases

### PR DESCRIPTION
**Pull request by madmax983**
🎯 Target: `sparse_squared_euclidean_distance` function in `src/core/vector/sparse.rs`.
💣 Risk: Edge cases like zero vectors, vectors with negative values, orthogonal vectors, and subnormal values could produce incorrect distances or panics if not explicitly verified. The stable sum of squared differences algorithm must be correct across these conditions.
🧪 Strategy: Added a robust table-driven unit test `test_sparse_squared_euclidean_distance_edge_cases` in `src/core/vector/tests.rs` to comprehensively test all these critical scenarios simultaneously with tight floating-point tolerance checks.
🔬 Verification: Run `cargo test --lib core::vector::tests::test_sparse_squared_euclidean_distance_edge_cases` to verify.

---
*PR created automatically by Jules for task [2622325317300753654](https://jules.google.com/task/2622325317300753654) started by @madmax983*

### CODE REVIEW

File: src/core/vector/tests.rs [hunk 1]
```diff
@@ -3015,3 +3015,70 @@ fn test_simd_mismatched_lengths_safety() {
...
+            b_values: vec![-1.0, 1.0],
+            expected: 4.0 + 4.0, // 8.0
+        },
+        TestCase {
+            name: "Subnormal values",
+            a_indices: vec![0],
+            a_values: vec![f32::MIN_POSITIVE],
+            b_indices: vec![0],
+            b_values: vec![-f32::MIN_POSITIVE],
+            expected: (2.0 * f32::MIN_POSITIVE) * (2.0 * f32::MIN_POSITIVE),
```
[the remark this answers is not in the document]
Comment (madmax983) [reply src/core/vector/tests.rs]: > **P2 Badge Replace MIN_POSITIVE with true subnormal inputs**
> 
> This case is labeled as a subnormal-value check, but `f32::MIN_POSITIVE` is the smallest **normal** float, and `(2.0 * f32::MIN_POSITIVE)^2` underflows to `0.0` in `f32`. That means the test effectively degenerates into another zero-distance assertion and won’t catch regressions in handling real subnormal magnitudes (around `1e-45`), creating false confidence about edge-case coverage.
> 
> Useful? React with 👍 / 👎.

@jules
